### PR TITLE
Repair "load indexes into memory" functions for web UI and SmartGraph edge collections

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Backport of https://github.com/arangodb/arangodb/pull/17478
Docs PR: https://github.com/arangodb/docs/pull/1166

Repair "Load indexes into memory" functionality in web UI, and make it work for SmartGraph edge collections, so that it actually starts the cache population.
The cache warmup is done in the background. The function will return as soon as all warmup tasks have been scheduled.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [x] Backport for 3.10: this PR
  - [x] Backport for 3.9: https://github.com/arangodb/arangodb/pull/17472
  - [ ] Backport for 3.8: -

#### Related Information

- [x] Docs PR: https://github.com/arangodb/docs/pull/1166
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/RocksDBEngine/RocksDBEdgeIndex.cpp [hunk 1]
```diff
@@ -73,17 +79,42 @@ using EdgeIndexCacheType = cache::TransactionalCache<cache::BinaryKeyHasher>;
...
       _lower(lower.data(), lower.size()),
       _upper(upper.data(), upper.size()) {}
 
-void RocksDBEdgeIndexWarmupTask::run() {
-  _index->warmupInternal(_trx, _lower, _upper);
+Result RocksDBEdgeIndexWarmupTask::run() {
+  DatabaseGuard databaseGuard(_databaseFeature, _dbName);
+  CollectionGuard collectionGuard(&databaseGuard.database(), _collectionName);
+
+  auto ctx = transaction::V8Context::CreateWhenRequired(
```
Comment (mpoeter) [inline arangod/RocksDBEngine/RocksDBEdgeIndex.cpp:96]: I think in the other PR we switched to a StandaloneContext.

File: arangod/RocksDBEngine/RocksDBEdgeIndex.cpp [hunk 2]
```diff
@@ -651,31 +682,49 @@ static std::string FindMedian(rocksdb::Iterator* it, std::string const& start,
...
+Result RocksDBEdgeIndex::scheduleWarmup() {
   if (!hasCache()) {
-    return;
+    return {};
   }
 
-  // prepare transaction for parallel read access
-  RocksDBTransactionState::toState(trx)->prepareForParallelReads();
+  auto ctx =
+      transaction::V8Context::CreateWhenRequired(_collection.vocbase(), false);
```
Comment (mpoeter) [inline arangod/RocksDBEngine/RocksDBEdgeIndex.cpp:691]: I think in the other PR we switched to a StandaloneContext.

**Review by mpoeter — COMMENTED** [review]
Two small comments, otherwise LGTM

**Review by mpoeter — APPROVED** [review]

**Review by neunhoef — APPROVED** [review]
LGTM

**Pull request merged**